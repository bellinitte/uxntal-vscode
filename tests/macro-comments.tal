%MACRO { x ( x ) x }
%MACRO { x (x x ) x }
%MACRO { x (x x )x ) x }
%MACRO { x (x) x )x()() x ) x }
%MACRO { x (x) x)X )x()() x ) x }
%MACRO { x (
    BRK LIT2kr @label/label &sublabel
    .label #12 #abcd $1 |100 ~lib.tal
    "string [ ]
) x }

%MACRO { X( x )X ) x }

( Nested comments )
%MACRO { x ( x ( x ) x ) x }
%MACRO { x (x x ( x ) x ) x }
%MACRO { x ( x ( x )x x ) x ) x }
%MACRO { x (x x ( x )x x ) x ) x }
%MACRO { x ( x ( x ( x ) x ) x ) x }

( Not nested -- inner comment has other characters after the `(` )
%MACRO { x ( x (x x ) x ) x }
%MACRO { x (x x (x x ) x ) x }
%MACRO { x (x x (x )x ) x } x
%MACRO { x (x x (x x ) x } x

( Unclosed comments )
%MACRO { x ( x } x
%MACRO { x (x x } x
%MACRO { x (x ( x ) x } x

( Nested macros )
%MACRO { x ( x %NESTED x ) x } x ( Nested macro definition gets highlighted )
%MACRO { x ( x ( x %NESTED x ) x ) x } x
%MACRO { x ( x %NESTED { x } x

( Macros in comments )
x ( x %MACRO { x } x ) x
x ( } ) x
