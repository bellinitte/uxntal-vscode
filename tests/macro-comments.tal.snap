>%MACRO { x ( x ) x }
#^ source.tal punctuation.rune.uxntal
# ^^^^^ source.tal meta.preprocessor.uxntal
#      ^ source.tal
#       ^ source.tal punctuation.braces.uxntal
#        ^ source.tal
#         ^ source.tal entity.name.function.uxntal
#          ^ source.tal
#           ^ source.tal comment.uxntal
#            ^ source.tal
#             ^ source.tal comment.uxntal
#              ^ source.tal
#               ^ source.tal comment.uxntal
#                ^ source.tal
#                 ^ source.tal entity.name.function.uxntal
#                  ^ source.tal
#                   ^ source.tal punctuation.braces.uxntal
>%MACRO { x (x x ) x }
#^ source.tal punctuation.rune.uxntal
# ^^^^^ source.tal meta.preprocessor.uxntal
#      ^ source.tal
#       ^ source.tal punctuation.braces.uxntal
#        ^ source.tal
#         ^ source.tal entity.name.function.uxntal
#          ^ source.tal
#           ^^ source.tal comment.uxntal
#             ^ source.tal
#              ^ source.tal comment.uxntal
#               ^ source.tal
#                ^ source.tal comment.uxntal
#                 ^ source.tal
#                  ^ source.tal entity.name.function.uxntal
#                   ^ source.tal
#                    ^ source.tal punctuation.braces.uxntal
>%MACRO { x (x x )x ) x }
#^ source.tal punctuation.rune.uxntal
# ^^^^^ source.tal meta.preprocessor.uxntal
#      ^ source.tal
#       ^ source.tal punctuation.braces.uxntal
#        ^ source.tal
#         ^ source.tal entity.name.function.uxntal
#          ^ source.tal
#           ^^ source.tal comment.uxntal
#             ^ source.tal
#              ^ source.tal comment.uxntal
#               ^ source.tal
#                ^^ source.tal comment.uxntal
#                  ^ source.tal
#                   ^ source.tal comment.uxntal
#                    ^ source.tal
#                     ^ source.tal entity.name.function.uxntal
#                      ^ source.tal
#                       ^ source.tal punctuation.braces.uxntal
>%MACRO { x (x) x )x()() x ) x }
#^ source.tal punctuation.rune.uxntal
# ^^^^^ source.tal meta.preprocessor.uxntal
#      ^ source.tal
#       ^ source.tal punctuation.braces.uxntal
#        ^ source.tal
#         ^ source.tal entity.name.function.uxntal
#          ^ source.tal
#           ^^^ source.tal comment.uxntal
#              ^ source.tal
#               ^ source.tal comment.uxntal
#                ^ source.tal
#                 ^^^^^^ source.tal comment.uxntal
#                       ^ source.tal
#                        ^ source.tal comment.uxntal
#                         ^ source.tal
#                          ^ source.tal comment.uxntal
#                           ^ source.tal
#                            ^ source.tal entity.name.function.uxntal
#                             ^ source.tal
#                              ^ source.tal punctuation.braces.uxntal
>%MACRO { x (x) x)X )x()() x ) x }
#^ source.tal punctuation.rune.uxntal
# ^^^^^ source.tal meta.preprocessor.uxntal
#      ^ source.tal
#       ^ source.tal punctuation.braces.uxntal
#        ^ source.tal
#         ^ source.tal entity.name.function.uxntal
#          ^ source.tal
#           ^^^ source.tal comment.uxntal
#              ^ source.tal
#               ^^^ source.tal comment.uxntal
#                  ^ source.tal
#                   ^^^^^^ source.tal comment.uxntal
#                         ^ source.tal
#                          ^ source.tal comment.uxntal
#                           ^ source.tal
#                            ^ source.tal comment.uxntal
#                             ^ source.tal
#                              ^ source.tal entity.name.function.uxntal
#                               ^ source.tal
#                                ^ source.tal punctuation.braces.uxntal
>%MACRO { x (
#^ source.tal punctuation.rune.uxntal
# ^^^^^ source.tal meta.preprocessor.uxntal
#      ^ source.tal
#       ^ source.tal punctuation.braces.uxntal
#        ^ source.tal
#         ^ source.tal entity.name.function.uxntal
#          ^ source.tal
#           ^ source.tal comment.uxntal
>    BRK LIT2kr @label/label &sublabel
#^^^^ source.tal
#    ^^^ source.tal comment.uxntal
#       ^ source.tal
#        ^^^^^^ source.tal comment.uxntal
#              ^ source.tal
#               ^^^^^^^^^^^^ source.tal comment.uxntal
#                           ^ source.tal
#                            ^^^^^^^^^ source.tal comment.uxntal
>    .label #12 #abcd $1 |100 ~lib.tal
#^^^^ source.tal
#    ^^^^^^ source.tal comment.uxntal
#          ^ source.tal
#           ^^^ source.tal comment.uxntal
#              ^ source.tal
#               ^^^^^ source.tal comment.uxntal
#                    ^ source.tal
#                     ^^ source.tal comment.uxntal
#                       ^ source.tal
#                        ^^^^ source.tal comment.uxntal
#                            ^ source.tal
#                             ^^^^^^^^ source.tal comment.uxntal
>    "string [ ]
#^^^^ source.tal
#    ^^^^^^^ source.tal comment.uxntal
#           ^ source.tal
#            ^ source.tal comment.uxntal
#             ^ source.tal
#              ^ source.tal comment.uxntal
>) x }
#^ source.tal comment.uxntal
# ^ source.tal
#  ^ source.tal entity.name.function.uxntal
#   ^ source.tal
#    ^ source.tal punctuation.braces.uxntal
>
>%MACRO { X( x )X ) x }
#^ source.tal punctuation.rune.uxntal
# ^^^^^ source.tal meta.preprocessor.uxntal
#      ^ source.tal
#       ^ source.tal punctuation.braces.uxntal
#        ^ source.tal
#         ^^ source.tal entity.name.function.uxntal
#           ^ source.tal
#            ^ source.tal entity.name.function.uxntal
#             ^ source.tal
#              ^^ source.tal entity.name.function.uxntal
#                ^ source.tal
#                 ^ source.tal entity.name.function.uxntal
#                  ^ source.tal
#                   ^ source.tal entity.name.function.uxntal
#                    ^ source.tal
#                     ^ source.tal punctuation.braces.uxntal
>
>( Nested comments )
#^ source.tal comment.uxntal
# ^ source.tal
#  ^^^^^^ source.tal comment.uxntal
#        ^ source.tal
#         ^^^^^^^^ source.tal comment.uxntal
#                 ^ source.tal
#                  ^ source.tal comment.uxntal
>%MACRO { x ( x ( x ) x ) x }
#^ source.tal punctuation.rune.uxntal
# ^^^^^ source.tal meta.preprocessor.uxntal
#      ^ source.tal
#       ^ source.tal punctuation.braces.uxntal
#        ^ source.tal
#         ^ source.tal entity.name.function.uxntal
#          ^ source.tal
#           ^ source.tal comment.uxntal
#            ^ source.tal
#             ^ source.tal comment.uxntal
#              ^ source.tal
#               ^ source.tal comment.uxntal
#                ^ source.tal
#                 ^ source.tal comment.uxntal
#                  ^ source.tal
#                   ^ source.tal comment.uxntal
#                    ^ source.tal
#                     ^ source.tal comment.uxntal
#                      ^ source.tal
#                       ^ source.tal comment.uxntal
#                        ^ source.tal
#                         ^ source.tal entity.name.function.uxntal
#                          ^ source.tal
#                           ^ source.tal punctuation.braces.uxntal
>%MACRO { x (x x ( x ) x ) x }
#^ source.tal punctuation.rune.uxntal
# ^^^^^ source.tal meta.preprocessor.uxntal
#      ^ source.tal
#       ^ source.tal punctuation.braces.uxntal
#        ^ source.tal
#         ^ source.tal entity.name.function.uxntal
#          ^ source.tal
#           ^^ source.tal comment.uxntal
#             ^ source.tal
#              ^ source.tal comment.uxntal
#               ^ source.tal
#                ^ source.tal comment.uxntal
#                 ^ source.tal
#                  ^ source.tal comment.uxntal
#                   ^ source.tal
#                    ^ source.tal comment.uxntal
#                     ^ source.tal
#                      ^ source.tal comment.uxntal
#                       ^ source.tal
#                        ^ source.tal comment.uxntal
#                         ^ source.tal
#                          ^ source.tal entity.name.function.uxntal
#                           ^ source.tal
#                            ^ source.tal punctuation.braces.uxntal
>%MACRO { x ( x ( x )x x ) x ) x }
#^ source.tal punctuation.rune.uxntal
# ^^^^^ source.tal meta.preprocessor.uxntal
#      ^ source.tal
#       ^ source.tal punctuation.braces.uxntal
#        ^ source.tal
#         ^ source.tal entity.name.function.uxntal
#          ^ source.tal
#           ^ source.tal comment.uxntal
#            ^ source.tal
#             ^ source.tal comment.uxntal
#              ^ source.tal
#               ^ source.tal comment.uxntal
#                ^ source.tal
#                 ^ source.tal comment.uxntal
#                  ^ source.tal
#                   ^^ source.tal comment.uxntal
#                     ^ source.tal
#                      ^ source.tal comment.uxntal
#                       ^ source.tal
#                        ^ source.tal comment.uxntal
#                         ^ source.tal
#                          ^ source.tal comment.uxntal
#                           ^ source.tal
#                            ^ source.tal comment.uxntal
#                             ^ source.tal
#                              ^ source.tal entity.name.function.uxntal
#                               ^ source.tal
#                                ^ source.tal punctuation.braces.uxntal
>%MACRO { x (x x ( x )x x ) x ) x }
#^ source.tal punctuation.rune.uxntal
# ^^^^^ source.tal meta.preprocessor.uxntal
#      ^ source.tal
#       ^ source.tal punctuation.braces.uxntal
#        ^ source.tal
#         ^ source.tal entity.name.function.uxntal
#          ^ source.tal
#           ^^ source.tal comment.uxntal
#             ^ source.tal
#              ^ source.tal comment.uxntal
#               ^ source.tal
#                ^ source.tal comment.uxntal
#                 ^ source.tal
#                  ^ source.tal comment.uxntal
#                   ^ source.tal
#                    ^^ source.tal comment.uxntal
#                      ^ source.tal
#                       ^ source.tal comment.uxntal
#                        ^ source.tal
#                         ^ source.tal comment.uxntal
#                          ^ source.tal
#                           ^ source.tal comment.uxntal
#                            ^ source.tal
#                             ^ source.tal comment.uxntal
#                              ^ source.tal
#                               ^ source.tal entity.name.function.uxntal
#                                ^ source.tal
#                                 ^ source.tal punctuation.braces.uxntal
>%MACRO { x ( x ( x ( x ) x ) x ) x }
#^ source.tal punctuation.rune.uxntal
# ^^^^^ source.tal meta.preprocessor.uxntal
#      ^ source.tal
#       ^ source.tal punctuation.braces.uxntal
#        ^ source.tal
#         ^ source.tal entity.name.function.uxntal
#          ^ source.tal
#           ^ source.tal comment.uxntal
#            ^ source.tal
#             ^ source.tal comment.uxntal
#              ^ source.tal
#               ^ source.tal comment.uxntal
#                ^ source.tal
#                 ^ source.tal comment.uxntal
#                  ^ source.tal
#                   ^ source.tal comment.uxntal
#                    ^ source.tal
#                     ^ source.tal comment.uxntal
#                      ^ source.tal
#                       ^ source.tal comment.uxntal
#                        ^ source.tal
#                         ^ source.tal comment.uxntal
#                          ^ source.tal
#                           ^ source.tal comment.uxntal
#                            ^ source.tal
#                             ^ source.tal comment.uxntal
#                              ^ source.tal
#                               ^ source.tal comment.uxntal
#                                ^ source.tal
#                                 ^ source.tal entity.name.function.uxntal
#                                  ^ source.tal
#                                   ^ source.tal punctuation.braces.uxntal
>
>( Not nested -- inner comment has other characters after the `(` )
#^ source.tal comment.uxntal
# ^ source.tal
#  ^^^ source.tal comment.uxntal
#     ^ source.tal
#      ^^^^^^ source.tal comment.uxntal
#            ^ source.tal
#             ^^ source.tal comment.uxntal
#               ^ source.tal
#                ^^^^^ source.tal comment.uxntal
#                     ^ source.tal
#                      ^^^^^^^ source.tal comment.uxntal
#                             ^ source.tal
#                              ^^^ source.tal comment.uxntal
#                                 ^ source.tal
#                                  ^^^^^ source.tal comment.uxntal
#                                       ^ source.tal
#                                        ^^^^^^^^^^ source.tal comment.uxntal
#                                                  ^ source.tal
#                                                   ^^^^^ source.tal comment.uxntal
#                                                        ^ source.tal
#                                                         ^^^ source.tal comment.uxntal
#                                                            ^ source.tal
#                                                             ^^^ source.tal comment.uxntal
#                                                                ^ source.tal
#                                                                 ^ source.tal comment.uxntal
>%MACRO { x ( x (x x ) x ) x }
#^ source.tal punctuation.rune.uxntal
# ^^^^^ source.tal meta.preprocessor.uxntal
#      ^ source.tal
#       ^ source.tal punctuation.braces.uxntal
#        ^ source.tal
#         ^ source.tal entity.name.function.uxntal
#          ^ source.tal
#           ^ source.tal comment.uxntal
#            ^ source.tal
#             ^ source.tal comment.uxntal
#              ^ source.tal
#               ^^ source.tal comment.uxntal
#                 ^ source.tal
#                  ^ source.tal comment.uxntal
#                   ^ source.tal
#                    ^ source.tal comment.uxntal
#                     ^ source.tal
#                      ^ source.tal entity.name.function.uxntal
#                       ^ source.tal
#                        ^ source.tal entity.name.function.uxntal
#                         ^ source.tal
#                          ^ source.tal entity.name.function.uxntal
#                           ^ source.tal
#                            ^ source.tal punctuation.braces.uxntal
>%MACRO { x (x x (x x ) x ) x }
#^ source.tal punctuation.rune.uxntal
# ^^^^^ source.tal meta.preprocessor.uxntal
#      ^ source.tal
#       ^ source.tal punctuation.braces.uxntal
#        ^ source.tal
#         ^ source.tal entity.name.function.uxntal
#          ^ source.tal
#           ^^ source.tal comment.uxntal
#             ^ source.tal
#              ^ source.tal comment.uxntal
#               ^ source.tal
#                ^^ source.tal comment.uxntal
#                  ^ source.tal
#                   ^ source.tal comment.uxntal
#                    ^ source.tal
#                     ^ source.tal comment.uxntal
#                      ^ source.tal
#                       ^ source.tal entity.name.function.uxntal
#                        ^ source.tal
#                         ^ source.tal entity.name.function.uxntal
#                          ^ source.tal
#                           ^ source.tal entity.name.function.uxntal
#                            ^ source.tal
#                             ^ source.tal punctuation.braces.uxntal
>%MACRO { x (x x (x )x ) x } x
#^ source.tal punctuation.rune.uxntal
# ^^^^^ source.tal meta.preprocessor.uxntal
#      ^ source.tal
#       ^ source.tal punctuation.braces.uxntal
#        ^ source.tal
#         ^ source.tal entity.name.function.uxntal
#          ^ source.tal
#           ^^ source.tal comment.uxntal
#             ^ source.tal
#              ^ source.tal comment.uxntal
#               ^ source.tal
#                ^^ source.tal comment.uxntal
#                  ^ source.tal
#                   ^^ source.tal comment.uxntal
#                     ^ source.tal
#                      ^ source.tal comment.uxntal
#                       ^ source.tal
#                        ^ source.tal entity.name.function.uxntal
#                         ^ source.tal
#                          ^ source.tal punctuation.braces.uxntal
#                           ^ source.tal
#                            ^ source.tal entity.name.function.uxntal
>%MACRO { x (x x (x x ) x } x
#^ source.tal punctuation.rune.uxntal
# ^^^^^ source.tal meta.preprocessor.uxntal
#      ^ source.tal
#       ^ source.tal punctuation.braces.uxntal
#        ^ source.tal
#         ^ source.tal entity.name.function.uxntal
#          ^ source.tal
#           ^^ source.tal comment.uxntal
#             ^ source.tal
#              ^ source.tal comment.uxntal
#               ^ source.tal
#                ^^ source.tal comment.uxntal
#                  ^ source.tal
#                   ^ source.tal comment.uxntal
#                    ^ source.tal
#                     ^ source.tal comment.uxntal
#                      ^ source.tal
#                       ^ source.tal entity.name.function.uxntal
#                        ^ source.tal
#                         ^ source.tal punctuation.braces.uxntal
#                          ^ source.tal
#                           ^ source.tal entity.name.function.uxntal
>
>( Unclosed comments )
#^ source.tal comment.uxntal
# ^ source.tal
#  ^^^^^^^^ source.tal comment.uxntal
#          ^ source.tal
#           ^^^^^^^^ source.tal comment.uxntal
#                   ^ source.tal
#                    ^ source.tal comment.uxntal
>%MACRO { x ( x } x
#^ source.tal punctuation.rune.uxntal
# ^^^^^ source.tal meta.preprocessor.uxntal
#      ^ source.tal
#       ^ source.tal punctuation.braces.uxntal
#        ^ source.tal
#         ^ source.tal entity.name.function.uxntal
#          ^ source.tal
#           ^ source.tal comment.uxntal
#            ^ source.tal
#             ^ source.tal comment.uxntal
#              ^ source.tal
#               ^ source.tal punctuation.braces.uxntal
#                ^ source.tal
#                 ^ source.tal entity.name.function.uxntal
>%MACRO { x (x x } x
#^ source.tal punctuation.rune.uxntal
# ^^^^^ source.tal meta.preprocessor.uxntal
#      ^ source.tal
#       ^ source.tal punctuation.braces.uxntal
#        ^ source.tal
#         ^ source.tal entity.name.function.uxntal
#          ^ source.tal
#           ^^ source.tal comment.uxntal
#             ^ source.tal
#              ^ source.tal comment.uxntal
#               ^ source.tal
#                ^ source.tal punctuation.braces.uxntal
#                 ^ source.tal
#                  ^ source.tal entity.name.function.uxntal
>%MACRO { x (x ( x ) x } x
#^ source.tal punctuation.rune.uxntal
# ^^^^^ source.tal meta.preprocessor.uxntal
#      ^ source.tal
#       ^ source.tal punctuation.braces.uxntal
#        ^ source.tal
#         ^ source.tal entity.name.function.uxntal
#          ^ source.tal
#           ^^ source.tal comment.uxntal
#             ^ source.tal
#              ^ source.tal comment.uxntal
#               ^ source.tal
#                ^ source.tal comment.uxntal
#                 ^ source.tal
#                  ^ source.tal comment.uxntal
#                   ^ source.tal
#                    ^ source.tal comment.uxntal
#                     ^ source.tal
#                      ^ source.tal punctuation.braces.uxntal
#                       ^ source.tal
#                        ^ source.tal entity.name.function.uxntal
>
>( Nested macros )
#^ source.tal comment.uxntal
# ^ source.tal
#  ^^^^^^ source.tal comment.uxntal
#        ^ source.tal
#         ^^^^^^ source.tal comment.uxntal
#               ^ source.tal
#                ^ source.tal comment.uxntal
>%MACRO { x ( x %NESTED x ) x } x ( Nested macro definition gets highlighted )
#^ source.tal punctuation.rune.uxntal
# ^^^^^ source.tal meta.preprocessor.uxntal
#      ^ source.tal
#       ^ source.tal punctuation.braces.uxntal
#        ^ source.tal
#         ^ source.tal entity.name.function.uxntal
#          ^ source.tal
#           ^ source.tal comment.uxntal
#            ^ source.tal
#             ^ source.tal comment.uxntal
#              ^ source.tal
#               ^ source.tal punctuation.rune.uxntal
#                ^^^^^^ source.tal meta.preprocessor.uxntal
#                      ^ source.tal
#                       ^ source.tal comment.uxntal
#                        ^ source.tal
#                         ^ source.tal comment.uxntal
#                          ^ source.tal
#                           ^ source.tal entity.name.function.uxntal
#                            ^ source.tal
#                             ^ source.tal punctuation.braces.uxntal
#                              ^ source.tal
#                               ^ source.tal entity.name.function.uxntal
#                                ^ source.tal
#                                 ^ source.tal comment.uxntal
#                                  ^ source.tal
#                                   ^^^^^^ source.tal comment.uxntal
#                                         ^ source.tal
#                                          ^^^^^ source.tal comment.uxntal
#                                               ^ source.tal
#                                                ^^^^^^^^^^ source.tal comment.uxntal
#                                                          ^ source.tal
#                                                           ^^^^ source.tal comment.uxntal
#                                                               ^ source.tal
#                                                                ^^^^^^^^^^^ source.tal comment.uxntal
#                                                                           ^ source.tal
#                                                                            ^ source.tal comment.uxntal
>%MACRO { x ( x ( x %NESTED x ) x ) x } x
#^ source.tal punctuation.rune.uxntal
# ^^^^^ source.tal meta.preprocessor.uxntal
#      ^ source.tal
#       ^ source.tal punctuation.braces.uxntal
#        ^ source.tal
#         ^ source.tal entity.name.function.uxntal
#          ^ source.tal
#           ^ source.tal comment.uxntal
#            ^ source.tal
#             ^ source.tal comment.uxntal
#              ^ source.tal
#               ^ source.tal comment.uxntal
#                ^ source.tal
#                 ^ source.tal comment.uxntal
#                  ^ source.tal
#                   ^ source.tal punctuation.rune.uxntal
#                    ^^^^^^ source.tal meta.preprocessor.uxntal
#                          ^ source.tal
#                           ^ source.tal comment.uxntal
#                            ^ source.tal
#                             ^ source.tal comment.uxntal
#                              ^ source.tal
#                               ^ source.tal comment.uxntal
#                                ^ source.tal
#                                 ^ source.tal comment.uxntal
#                                  ^ source.tal
#                                   ^ source.tal entity.name.function.uxntal
#                                    ^ source.tal
#                                     ^ source.tal punctuation.braces.uxntal
#                                      ^ source.tal
#                                       ^ source.tal entity.name.function.uxntal
>%MACRO { x ( x %NESTED { x } x
#^ source.tal punctuation.rune.uxntal
# ^^^^^ source.tal meta.preprocessor.uxntal
#      ^ source.tal
#       ^ source.tal punctuation.braces.uxntal
#        ^ source.tal
#         ^ source.tal entity.name.function.uxntal
#          ^ source.tal
#           ^ source.tal comment.uxntal
#            ^ source.tal
#             ^ source.tal comment.uxntal
#              ^ source.tal
#               ^ source.tal punctuation.rune.uxntal
#                ^^^^^^ source.tal meta.preprocessor.uxntal
#                      ^ source.tal
#                       ^ source.tal comment.uxntal
#                        ^ source.tal
#                         ^ source.tal comment.uxntal
#                          ^ source.tal
#                           ^ source.tal punctuation.braces.uxntal
#                            ^ source.tal
#                             ^ source.tal entity.name.function.uxntal
>
>( Macros in comments )
#^ source.tal comment.uxntal
# ^ source.tal
#  ^^^^^^ source.tal comment.uxntal
#        ^ source.tal
#         ^^ source.tal comment.uxntal
#           ^ source.tal
#            ^^^^^^^^ source.tal comment.uxntal
#                    ^ source.tal
#                     ^ source.tal comment.uxntal
>x ( x %MACRO { x } x ) x
#^ source.tal entity.name.function.uxntal
# ^ source.tal
#  ^ source.tal comment.uxntal
#   ^ source.tal
#    ^ source.tal comment.uxntal
#     ^ source.tal
#      ^^^^^^ source.tal comment.uxntal
#            ^ source.tal
#             ^ source.tal comment.uxntal
#              ^ source.tal
#               ^ source.tal comment.uxntal
#                ^ source.tal
#                 ^ source.tal comment.uxntal
#                  ^ source.tal
#                   ^ source.tal comment.uxntal
#                    ^ source.tal
#                     ^ source.tal comment.uxntal
#                      ^ source.tal
#                       ^ source.tal entity.name.function.uxntal
>x ( } ) x
#^ source.tal entity.name.function.uxntal
# ^ source.tal
#  ^ source.tal comment.uxntal
#   ^ source.tal
#    ^ source.tal comment.uxntal
#     ^ source.tal
#      ^ source.tal comment.uxntal
#       ^ source.tal
#        ^ source.tal entity.name.function.uxntal
>